client/src/util/fetchFromAPI.test.js

```diff
@@ -1,3 +1,14 @@
-import { it } from "vitest";
+import { it, expect } from "vitest";
+import { queryAPI } from "./fetchFromAPI";
 
-it("should fetch all items");
+//Mocking functions
+it("verify if food items are edited when user clicks button", async () => {
+  queryAPI([]);
+});
+expect(fetch).toHaveBeenCalledWith("http://localhost:5000/menu_items", {
+  method: "GET",
+  body: JSON.stringify,
+  headers: {
+    "Content-Type": "application/json",
+  },
+});
```

Comment:
Thanks for putting this up, it's helpful to see where you're mind is at.  What you have here is something that tests the _implementation details_ of your `queryAPI` function and does not test any of the _behavior_.  This is something you'll want to train yourself to identify and avoid.  If you'd like a deep dive into why this is bad, [here's an excellent article from Kent Dodds](https://kentcdodds.com/blog/testing-implementation-details).  

Don't feel compelled to read that right now, though.  The core point is that we want to test the things that would be bad if they broke.  We do _not_ want to test the tools we used to get there; a.k.a. the implementation.  The fact that you used `fetch` to make the call is an implementation detail.  You could have used some helper library, or not used fetch at all.  a `behavior` on the other hand, is something this function is guaranteed to do regardless of what is used under-the-hood to accomplish it.  An example of a behavior is that an `onSuccess` callback will be fired if the API request is successful.

With all of that said, it's clear I've done you a disservice by not giving you three concrete things to test about this utility function; I left you to sort that out for yourself which isn't very mentor-like of me.  My apologies.

Here are three behaviors of this utility that I'd like you to write tests for.

1. If the `fetch` call results in a 200 response, the `onSuccess` callback is fired
2. If the `fetch` call results in a non-200 response the `onError` callback is fired
3. If the `method` is `POST` a `body` option cannot be "nullish" (as in, `null` or `undefined`) and must be serializable to JSON.

One thing you'll want to do in order to accomplish this is to mock out the `fetch` function since we don't _actually_ wish to make calls to the API during testing.  Have a look at [fetch-mock](https://www.npmjs.com/package/fetch-mock)

Reply:
<img width="453" alt="Screen Shot 2023-01-23 at 1 21 21 PM" src="https://user-images.githubusercontent.com/34128735/214119065-c5f059d4-38b5-488a-85d8-9adbaaf1d5b3.png">

Reply:
Bookmarked that article @brentmclark thank you .